File: solutions/ocaml/difference-of-squares/1/difference_of_squares.ml
let sum (arr : int list) : int = List.fold_left (fun acc a -> acc + a) 0 arr

let square_of_sum n = 
    (sum (List.init n (fun i -> i + 1))) * (sum (List.init n (fun i -> i + 1)))

let sum_of_squares (n : int) = sum (List.map (fun x -> x*x) (List.init n (fun i -> i + 1)))
    

let difference_of_squares n =
    square_of_sum n - sum_of_squares n

File: solutions/zig/difference-of-squares/1/difference_of_squares.zig
pub fn squareOfSum(number: usize) usize {
    var ans: usize = 0;
    for (1..number + 1) |i| {
        ans += i;
    }
    ans *= ans;
    return ans;
}

pub fn sumOfSquares(number: usize) usize {
    var ans: usize = 0;
    for (1..number + 1) |i| {
        ans += i * i;
    }
    return ans;
}

pub fn differenceOfSquares(number: usize) usize {
    return squareOfSum(number) - sumOfSquares(number);
}
